motor/discovery/k8s/list_workloads.go

```diff
@@ -180,7 +180,15 @@ func ListNamespacedObj[T runtime.Object](
 		obj, _ := meta.Accessor(workloads[i])
 		log.Debug().Str("name", obj.GetName()).Str("connection", asset.Connections[0].Host).Msgf("resolved %s", workloadType)
 
-		assets = append(assets, asset)
+		assetsIdx[asset.PlatformIds[0]] = asset
+	}
+
+	// Return a unique list of assets. Manifests can contain a namespaces that is an empty string. When we try to list k8s
+	// resources for the empty namespace, that actually means list all resources. Therefore we can have duplicate entries in the list.
+	// Here we just return only the unique assets to make sure the code works correctly with both manifests and k8s API.
```

Comment:
Because of this, I was curious to give this command a try: `cnquery run k8s /tmp/2cronjobs.yaml --namespaces default -c 'k8s.cronjob.name'`. It fails with:
```
x could not resolve asset error="failed to create asset from namespace: could not determine object kind " asset=
FTL failed to run query error="could not find an asset that we can connect to"
```
Without the namespace filter, cnquery discovers both CronJobs.

I'm not sure whether this is really a valid use-case for the manifest scan. Perhaps something, we could fix in a follow up?

For reference the file I used: https://gist.github.com/czunker/ee29f077a0babcf6f2ce847c29823e8c